### PR TITLE
Show the context menu when we long press assignee of a task

### PR DESCRIPTION
**Pull request by ShogunFire**
### Details
Like for other previews, we show the context menu when we long press a task assignee

### Fixed Issues

$ https://github.com/Expensify/App/issues/23026
PROPOSAL: https://github.com/Expensify/App/issues/23026#issuecomment-1646266288

### Tests
1. Click FAB -> Assign task
2. Add task details
3. Select assignee
4. Long press on Assignee of task
5. Verify that the context menu shows on platforms with touch screen 

- [x] Verify that no errors appear in the JS console

### Offline tests

Same tests

### QA Steps

Same tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/19537677/bc797587-a360-4a96-a1ea-46cb8e43980a

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/19537677/ae8968eb-d6d2-437a-8d28-0a8efd41fc39

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/19537677/100f9768-e0f3-4a28-8271-35b3be072a26

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/19537677/f32e7398-38f7-4937-a14d-a28780da35b2

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/19537677/00f1935d-6c75-4736-9395-fdb20c83792b

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/19537677/baef6c94-8232-4601-bbd9-553c36f620ce

</details>

### CODE REVIEW

**Comment by abdulrahuman5196** [comment]
@ShogunFire The author's checklist is pending on this

**Comment by ShogunFire** [comment]
I will try to add the videos today

**Comment by abdulrahuman5196** [comment]
Any update on the PR @ShogunFire ?

**Comment by ShogunFire** [comment]
I have troubles testing on android because of a reanimated error, on mweb and safari it works but on ios it doesn't, I am not sure why but it might be because we use onPress in TextLink which is used in MentionRenderer, we also have a preventDefault in the callback of onPress here that could cause this problem https://github.com/Expensify/App/blob/c228338a2768f28eb9a2516e2a5de0f02f720d03/src/components/TextLink.js#L47

**Comment by abdulrahuman5196** [comment]
@ShogunFire Any update on the issue?

**Comment by ShogunFire** [comment]
I have more informations on the cause: 
In MentionUserRenderer we pass onPress to the TextLink (which will pass it to the Text component)
For android and ios, react native uses a Touchable when we pass the prop onPress to the Text component.

When there is a nested Touchable or Pressable, only the nested events are called and the events from the parent Pressable are ignored, that's why the long press event is never called and the context menu doesn't show.

I am trying to find a solution but so far it seems difficult unless we don't pass the onPress when the user mention is in a task preview. The consequence of this is that in task previews when we click the user, it will open the task report instead of the user details, is it acceptable ?

**Comment by abdulrahuman5196** [comment]
@ShogunFire I don't understand the issue you are saying?

Can you provide code pointer or videos to clarify?

**Comment by ShogunFire** [comment]
[Here](https://github.com/Expensify/App/blob/eb985defedccd75ff4d347cff2c7adb433e068d6/src/components/ReportActionItem/TaskPreview.js#L107) I set up the onLongPress event to show the context menu.

But it doesn't work when we click on the user mention. We put the user-mention tag [here](https://github.com/Expensify/App/blob/eb985defedccd75ff4d347cff2c7adb433e068d6/src/components/ReportActionItem/TaskPreview.js#L94) 

And the text inside that tag is rendered by [MentionUserRenderer](https://github.com/Expensify/App/blob/eb985defedccd75ff4d347cff2c7adb433e068d6/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.js) that uses onPress event [here](https://github.com/Expensify/App/blob/eb985defedccd75ff4d347cff2c7adb433e068d6/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.js#L71)

Basically for ios and android the Pressable that is used is the one in MentionUserRenderer and not the one in TaskPreview so the longPress event is never called (the context menu doesn't show)

What I can try to do is pass arguments to user-mention tag to be able to set up the onLongPress event directly in MentionUserRenderer

**Comment by abdulrahuman5196** [comment]
@ShogunFire This is already working for normally messages, how is it working there? Can we re-use the same logic here as well?

**Comment by ShogunFire** [comment]
Normal messages don't have onPress event. User mentions show the Details of the user when you click them.

**Comment by ShogunFire** [comment]
@abdulrahuman5196 Sorry for the late update, so another PR tried to solve the issue with user mention here: https://github.com/Expensify/App/pull/29648 . To work in our case it was just lacking a provider around TaskPreview which I added. Now everything should work fine. I am finishing the videos, other than that this is ready for review

**Comment by ShogunFire** [comment]
@abdulrahuman5196 I resolved the conflicts

**Comment by abdulrahuman5196** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>







https://github.com/Expensify/App/assets/46707890/e9b487db-05c2-4709-bc97-298a1b9a127a



</details>

<details>
<summary>Android: mWeb Chrome</summary>




https://github.com/Expensify/App/assets/46707890/ee3154c4-7248-47c9-8d8d-d5de0e0d8651


</details>

<details>
<summary>iOS: Native</summary>







https://github.com/Expensify/App/assets/46707890/81571274-2a3a-4854-80f8-e2562b62678c



</details>

<details>
<summary>iOS: mWeb Safari</summary>






https://github.com/Expensify/App/assets/46707890/f131a9ef-0ce5-4253-b947-bab352ab28ea



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>





https://github.com/Expensify/App/assets/46707890/34a81636-0c17-4a73-8bfc-61dd34f021eb

</details>

<details>
<summary>MacOS: Desktop</summary>





https://github.com/Expensify/App/assets/46707890/81064bf8-2e9d-470b-9e46-4e9901be1895



</details>

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @AndrewGable 

🎀 👀 🎀
C+ Reviewed

**Comment by abdulrahuman5196** [comment]
Gentle ping @AndrewGable

**Review by AndrewGable — APPROVED** [review]
